src/GraphCoarsen.cpp

```diff
@@ -461,23 +483,26 @@ void GraphCoarsen::BuildPEdges(
 
     Agg_cdof_edge_ = agg_dof_builder.GetAgg_cdof_edge(nAggs, total_num_traces + bubble_counter);
 
-    auto elem_mbuilder_ptr = dynamic_cast<ElementMBuilder*>(&mbuilder);
+    auto elem_mbuilder_ptr = dynamic_cast<ElementMBuilder*>(&coarse_mbuilder);
     if (elem_mbuilder_ptr)
     {
         elem_mbuilder_ptr->SetAggToEdgeDofsTableReference(*Agg_cdof_edge_);
     }
 
     mfem::SparseMatrix face_Agg(smoothg::Transpose(Agg_face));
-    // next line assume M_proc_ is diagonal
-    mfem::Vector M_v(M_proc_.GetData(), M_proc_.Width()), Mloc_v;
-    mbuilder.ResetEdgeCdofMarkers(total_num_traces + bubble_counter);
 
+    auto edge_vert = smoothg::Transpose(D_proc_); // TODO: use vertex_edge
```

Comment:
I get nervous about TODO comments in PRs. Either fix it or document exactly what's wrong?

Reply:
Right now this is fine. I was thinking this will be wrong for multilevel (although in multilevel many things in this code will be wrong anyway). I will just erase this TODO.